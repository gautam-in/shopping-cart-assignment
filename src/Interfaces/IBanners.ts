export default interface IBanners {
    id: string;
    order: number;
    isActive: boolean;
    bannerImageUrl: string;
    bannerImageAlt: string;
}
