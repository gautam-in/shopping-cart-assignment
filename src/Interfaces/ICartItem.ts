export interface ICartItem {
    id: string;
    name: string;
    imageURL: string;
    price: number;
    unit: number;
}
