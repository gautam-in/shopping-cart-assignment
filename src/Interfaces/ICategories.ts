export interface ICategories {
    id: string;
    order: number;
    enabled: boolean;
    key: string;
    name: string;
    description: string;
    imageUrl: string;
}
