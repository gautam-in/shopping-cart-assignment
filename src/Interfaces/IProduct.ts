export interface IProduct {
    id: string;
    name: string;
    imageURL: string;
    description: string;
    price: number;
    stock: number;
    category: string;
    sku: string;
}
