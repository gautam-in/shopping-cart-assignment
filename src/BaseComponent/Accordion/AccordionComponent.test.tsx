import { ShallowWrapper, shallow } from "enzyme";
import { ICategories } from "../../Interfaces/ICategories";
import {
    AccordionComponent,
    IAccordionComponentProps,
} from "./AccordionComponent";

describe("AccordionComponent", () => {
    const categorie:ICategories = {
        id:'0',
        order: 0,
        enabled: true,
        key: '0',
        name: 'name',
        description: 'name',
        imageUrl: 'name',
    }
    const categoriesListProps:Array<ICategories> = [categorie];
    const tree: ShallowWrapper<IAccordionComponentProps> = shallow(
        <AccordionComponent categoriesList={categoriesListProps} selectedCategorieID={null} />
    );

    test("AccordionComponent Snapshot render", () => {
        expect(tree).toMatchSnapshot();
    });

    test("Render Child AccordionComponent based on props", () => {
        expect(tree.children().length).toEqual(categoriesListProps.length)
    });

    test("Do Not Render Child AccordionComponent if categorie is not enable", () => {
        const categorie:ICategories = {
            id:'0',
            order: 0,
            enabled: false,
            key: '0',
            name: 'name',
            description: 'name',
            imageUrl: 'name',
        }
        const categoriesListProps:Array<ICategories> = [categorie];
        const tree: ShallowWrapper<IAccordionComponentProps> = shallow(
            <AccordionComponent categoriesList={categoriesListProps} selectedCategorieID={null} />
        );
        expect(tree.children().length).toEqual(0)
    });
});