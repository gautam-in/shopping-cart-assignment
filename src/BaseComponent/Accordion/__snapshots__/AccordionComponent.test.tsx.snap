// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionComponent AccordionComponent Snapshot render 1`] = `ShallowWrapper {}`;
